cpp/unreal_plugins/CoreUtils/Source/CoreUtils/InputActionComponent.h

```diff
@@ -0,0 +1,112 @@
+{
+    GENERATED_BODY()
+public:
+    UInputActionComponent()
+    {
+        SP_LOG_CURRENT_FUNCTION();
+
+        PrimaryComponentTick.bCanEverTick = true;
+        PrimaryComponentTick.bTickEvenWhenPaused = false;
+    }
+
+    virtual ~UInputActionComponent()
+    {
+        SP_LOG_CURRENT_FUNCTION();
+
+        input_component_ = nullptr;
+        input_action_descs_.clear();
+    }
+
+    // USceneComponent interface
+    void TickComponent(float delta_time, ELevelTick level_tick, FActorComponentTickFunction* this_tick_function) override
+    {
+        USceneComponent::TickComponent(delta_time, level_tick, this_tick_function);
+
+        if (input_component_ && apply_input_action_func_) {
+            for (auto& input_action_desc : input_action_descs_) {
+                float axis_value = input_component_->GetAxisValue(Unreal::toFName(input_action_desc.axis_));
+                if (axis_value >= input_action_desc.threshold_) {
+                    apply_input_action_func_(input_action_desc.key_);
+                }   
+            }
+        }
+    }
+
+    // Must be called in BeginPlay() or later as GetWorld() needs to be valid.
+    template <typename T>
+    void bindInputActions(const std::map<std::string, T>& input_actions)
+    {
+        SP_ASSERT(GetWorld());
+        SP_ASSERT(GetWorld()->GetFirstPlayerController());
+
+        input_component_ = GetWorld()->GetFirstPlayerController()->InputComponent;
+        SP_ASSERT(input_component_);
+
+        UPlayerInput* player_input = GetWorld()->GetFirstPlayerController()->PlayerInput;
+        SP_ASSERT(player_input);
+
+        for (auto& input_action : input_actions) {
+            InputActionDesc input_action_desc;
+            input_action_desc.key_ = input_action.first;
+
+            bool include_actor_name = true;
+            std::string separator = ".";
+            input_action_desc.axis_ =
+                Unreal::getFullyQualifiedComponentName(this, separator, include_actor_name) + "." + input_action_desc.key_;
+
+            player_input->AddAxisMapping(FInputAxisKeyMapping(
+                Unreal::toFName(input_action_desc.axis_),
+                FKey(Unreal::toFName(input_action_desc.key_)),
+                input_action_desc.scale_));
+
+            input_component_->BindAxis(Unreal::toFName(input_action_desc.axis_));
+            
+            input_action_descs_.push_back(input_action_desc);
+        }
+    }
+
+    // Set by user code to specify what happens when an action is triggered, e.g., UUrdfRobotComponent and UUrdfJointComponent.
+    std::function<void(const std::string&)> apply_input_action_func_;
+
+private:
+    struct InputActionDesc
+    {
+        std::string key_;
+        std::string axis_;
+        float scale_ = 1.0f;
+        float threshold_ = 1.0f;
+    };
+
+    // Set in setPlayerInputActions(...), but we expose the underlying variable as public in case user code has an alternative way of setting it.
```

Comment:
No need for either of these comments because they are: (a) out-of-date and therefore misleading; and (b) private so readers don't need to understand them in as much detail as the public interface.

Reply:
Updated code